pkg/kubelet/container/interface.go

```diff
@@ -0,0 +1,335 @@
+// The prefix is optional.  If the prefix is not specified, the key is assumed to be private
+// to the user.  Other system components that wish to use labels must specify a prefix.  The
+// "kubernetes.io/" prefix is reserved for use by kubernetes components.
+type Labels map[string]string
+
+// LinuxPodSandboxConfig holds platform-specific configuraions for Linux
+// host platforms and Linux-based containers.
+type LinuxPodSandboxConfig struct {
+	// CgroupParent is the parent cgroup of the sandbox.
+	CgroupParent string
+	// NamespaceOptions contains configurations for the sandbox's namespaces.
+	// This will be used only if the PodSandbox uses namespace for isolation.
+	NamespaceOptions NamespaceOptions
+}
+
+// NamespaceOptions provides options for Linux namespaces.
+type NamespaceOptions struct {
+	// HostNetwork uses the host's network namespace.
+	HostNetwork bool
+	// HostPID uses the host's pid namesapce.
+	HostPID bool
+	// HostIPC uses the host's ipc namespace.
+	HostIPC bool
+}
+
+// DNSOptions specifies the DNS servers and search domains.
+type DNSOptions struct {
+	// Servers is a list of DNS servers of the cluster.
+	Servers []string
+	// Searches is a list of DNS search domains of the cluster.
+	Searchs []string
+}
+
+type PodSandboxState string
+
+const (
+	// PodSandboxActive means the sandbox is functioning properly.
+	PodSandboxActive PodSandboxState = "active"
+	// PodSandboxInactive means the sandbox is not functioning properly.
+	PodSandboxInactive PodSandboxState = "inactive"
+)
+
+// PodSandboxFilter is used to filter a list of PodSandboxes.
+type PodSandboxFilter struct {
+	// Name of the sandbox.
+	Name string
+	// ID of the sandbox.
+	ID string
+	// State of the sandbox.
+	State string
+	// LabelSelector to select matches.
+	LabelSelector unversioned.LabelSelector
+}
+
+// PodSandboxListItem contains minimal information about a sandbox.
+type PodSandboxListItem struct {
+	ID    string
+	State PodSandboxState
+}
+
+// PodSandboxStatus contains the status of the PodSandbox.
+type PodSandboxStatus struct {
+	// ID of the sandbox.
+	ID string
+	// State of the sandbox.
+	State PodSandboxState
+	// Network contains network status if network is handled by the runtime.
+	Network *PodSandboxNetworkStatus
+	// Status specific to a Linux sandbox.
+	Linux *LinuxPodSandboxStatus
+}
+
+// PodSandboxNetworkStatus is the status of the network for a PodSandbox.
+type PodSandboxNetworkStatus struct {
+	IP string
+}
+
+// Namespaces contains paths to the namespaces.
+type Namespaces struct {
+	// Network is the path to the network namespace.
+	Network string
+}
+
+// LinuxSandBoxStatus contains status specific to Linux sandboxes.
+type LinuxPodSandboxStatus struct {
+	// Namespaces contains paths to the sandbox's namespaces.
+	Namespaces *Namespaces
+}
+
+// PodSandboxResources contains the CPU/memory resource requirements.
+type PodSandboxResources struct {
+	// CPU resource requirement.
+	CPU api.ResourceRequirements
```

Comment:
AFAIK, VMs need to be specified limits and not requests, unless they can
baloon. I hope that runtimes will not use `requests` as `limits`.

On Wed, May 18, 2016 at 5:21 PM, Yu-Ju Hong <notifications@github.com>
wrote:

> In pkg/kubelet/container/interface.go
> <https://github.com/yujuhong/kubernetes/pull/1#discussion_r63805773>:
>
> > +// Namespaces contains paths to the namespaces.
> > +type Namespaces struct {
> > +	// Network is the path to the network namespace.
> > +	Network string
> > +}
> > +
> > +// LinuxSandBoxStatus contains status specific to Linux sandboxes.
> > +type LinuxPodSandboxStatus struct {
> > +	// Namespaces contains paths to the sandbox's namespaces.
> > +	Namespaces *Namespaces
> > +}
> > +
> > +// PodSandboxResources contains the CPU/memory resource requirements.
> > +type PodSandboxResources struct {
> > +	// CPU resource requirement.
> > +	CPU api.ResourceRequirements
>
> If there are no limits, shouldn't required be used to create a VM, for
> example?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/yujuhong/kubernetes/pull/1/files/0ca17bb20ed69d11610d8134406ee88712444ce9#r63805773>
>